ir/src/integrity_stmts/graph.rs

```diff
@@ -184,6 +187,34 @@ impl AlgebraicGraph {
         }
     }
 
+    /// Adds a trace column to the graph and returns the node index and trace segment.
+    ///
+    /// # Errors
+    /// Returns an error if the identifier is bigger than overall number of columns in segment.
+    fn insert_column_access(
+        &mut self,
+        symbol_table: &SymbolTable,
+        column_access: ColumnAccess,
+    ) -> Result<(TraceSegment, NodeIndex), SemanticError> {
+        if column_access.column_idx() as u16
+            >= symbol_table.segment_width()[column_access.segment_idx()]
+        {
+            return Err(SemanticError::IndexOutOfRange(format!(
+                "Out-of-range index {} in trace segment {} of length {}",
+                column_access.column_idx(),
+                column_access.segment_idx(),
+                symbol_table.segment_width()[column_access.segment_idx()]
+            )));
+        }
+        let trace_access = TraceAccess::new(
+            column_access.segment_idx() as u8,
+            column_access.column_idx(),
+            column_access.row_offset(),
+        );
```

Comment:
Since these contain all the same information, can you implement the `From` trait instead?

Also, we should probably check the size of segment_idx somewhere before casting it.